xbmc/cores/RetroPlayer/rendering/RPRenderManager.cpp

```diff
@@ -690,3 +691,44 @@ CRenderVideoSettings CRPRenderManager::GetEffectiveSettings(
 
   return effectiveSettings;
 }
+
+void CRPRenderManager::SaveThumbnail(const std::string& path)
+{
+  m_bufferMutex.lock();
+
+  if (m_renderBuffers.size() <= 0)
+  {
+    m_bufferMutex.unlock();
+    return;
+  }
+
+  const uint8_t* const data = m_renderBuffers[0]->GetMemory();
+  const size_t size = m_renderBuffers[0]->GetFrameSize();
+  const unsigned int width = m_renderBuffers[0]->GetWidth();
+  const unsigned int height = m_renderBuffers[0]->GetHeight();
+  const AVPixelFormat format = m_renderBuffers[0]->GetFormat();
+
+  std::unique_ptr<uint8_t> copiedData(new uint8_t[size]);
+  std::memcpy(copiedData.get(), data, size);
+
+  m_bufferMutex.unlock();
+
+  const int stride = CRenderTranslator::TranslateWidthToBytes(width, format);
+
+  unsigned int scaleWidth = 400;
+  unsigned int scaleHeigth = 220;
+  CPicture::GetScale(width, height, scaleWidth, scaleHeigth);
+
+  const int bytesPerPixel = 4;
+  std::unique_ptr<uint8_t> scaledImage(new uint8_t[scaleWidth * scaleHeigth * bytesPerPixel]);
+
+  const AVPixelFormat outFormat = AV_PIX_FMT_BGR0;
```

Comment:
What's the reason for using `AV_PIX_FMT_BGR0` ??

Reply:
The first emulator that I tested this with used this pixel format and emulators with other pixel formats didn't work (the thumbnail was messed up) so in a talk I had with @VelocityRa we decided to transform all pixel format to `AV_PIX_FMT_BGR0 `